### PR TITLE
[Data] Correctly refresh elapsed time in `ProgressBar` output

### LINKED ISSUE #44689: [Data] Progress bar doesn't refresh if slow UDF
### What happened + What you expected to happen

I'm running a Ray Data job with a very slow UDF (15+ minutes per block). The progress bar doesn't update, so I'm not sure if Ray Data is just waiting for outputs, or if my program has actually hanged.

### Versions / Dependencies

905ca61afb1e24b876f4c8c17b78c192013b13e4

### Reproduction script

```python
import time

import ray


def sleep(batch):
    time.sleep(60)
    return batch


ray.data.range(10, override_num_blocks=10).map_batches(
    sleep, batch_size=None
).materialize()
```

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by scottjlee**
## Why are these changes needed?

When using Ray Data with a slow UDF for map operators, the progress bar's "elapsed time" item is only updated each time a task finishes. If the task is slow, this means that the elapsed time is left "hanging" and appears to be stuck, when this is not necessarily the case.

This PR fixes the refreshing of the progress bar, so that the elapsed time ticks up as expected. For the following example code, we can see the change in behavior in the ticking of the progress bar:
```
import time
import ray

def f(batch):
    time.sleep(10)
    return batch

ds = ray.data.range(50, override_num_blocks=50).map(f)
ds.materialize()
```

- Before (ticks every 10 seconds):
https://github.com/user-attachments/assets/d46f0d6f-dacc-4148-a12c-001fcf44f008

- After (ticks every 1 second):
https://github.com/user-attachments/assets/88807215-9e81-434f-b186-b5d597bc3c59

## Related issue number

Closes https://github.com/ray-project/ray/issues/44689

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by omatthew98 — APPROVED** [review]
nice

File: python/ray/data/_internal/execution/streaming_executor.py [hunk 1]
```diff
@@ -320,6 +320,9 @@ def _scheduling_loop_step(self, topology: Topology) -> bool:
         # Update the progress bar to reflect scheduling decisions.
         for op_state in topology.values():
             op_state.refresh_progress_bar(self._resource_manager)
+        # Refresh the global progress bar to update elapsed time progress.
+        if self._global_info and self._global_info._bar:
+            self._global_info._bar.refresh()
```
Comment (bveeramani) [inline python/ray/data/_internal/execution/streaming_executor.py:325]: Should we add a `refresh` method to `ProgressBar`? My impression is that this is breaking an abstraction barrier by accessing the private `ProgressBar._bar` attribute?

**Review by bveeramani — APPROVED** [review]
Nice

File: python/ray/data/_internal/execution/streaming_executor.py (hunk 1 above)
Comment (scottjlee) [reply python/ray/data/_internal/execution/streaming_executor.py:325]: yeah good point, that will help clean up the logic as well. added it

**Pull request merged**